### ISSUE TITLE
Controller Error, effects disappeared/not loading

### ISSUE THREAD

**Issue by Valachord** [polychromatic/polychromatic#321]
After rebooting my machine earlier, the effects section suddenly doesn't load any of my saved effects, they don't show up in the sidebar.

It seems that I can create new effects, but I haven't tried saving any as I don't wanna risk somehow overwriting my already, non-showing, saved effects.

The app also stopped autostarting, don't know if it is directly related, but it happened after the same reboot.

I'm getting the message:

"Controller Error"

"Eeks! An uncaught exception was thrown while loading the tab.
The application will attempt to continue, but get ready in case something breaks again!
This may be reported as a bug on the project's issue tracker."
```
"Traceback (most recent call last):
  File ’/usr/bin/polychromatic-controller’, line 369, in _change_tab
    tab_objects[index].set_tab()
  File ’/usr/lib/python3.9/site-packages/polychromatic/controller/shared.py’, line 1352, in set_tab
    file_list = self.fileman.get_item_list()
  File ’/usr/lib/python3.9/site-packages/polychromatic/fileman.py’, line 181, in get_item_list
    items.append(self._get_parsed_keys(data, path))
  File ’/usr/lib/python3.9/site-packages/polychromatic/fileman.py’, line 162, in _get_parsed_keys
    parsed[’name’] = self._get_i18n_key(data, ’name’)
  File ’/usr/lib/python3.9/site-packages/polychromatic/fileman.py’, line 100, in _get_i18n_key
    if localized_key_full in data.keys():
AttributeError: ’NoneType’ object has no attribute ’keys’"
```
I'm using Arch (btw) :sweat_smile:

**Comment by lah7** [comment]
Saving new effects never overwrites any old ones (even with the same name). They're stored in `~/.config/polychromatic/effects/`. It looks like one of your effects contains invalid data and breaks the application when it's trying to load them. Likely, this is why the tray applet doesn't start any more.

What locale is your system out of interest?

    echo $LANG

Try running from the Terminal, this should list the problematic file when entering that tab - please let me know the output:

    polychromatic-controller -v

If you don't mind, please could you upload the offending JSON file(s) here so I can take a closer look? In the meantime, moving the problematic file outside of the `effects` folder should restore functionality until the root issue is found.

**Comment by lah7** [comment]
I've fixed the root cause, so the effects tab will not halt ⛔ after failing to process one of the effects.

I'm still curious to know how one of the effects ended up with _bad_ data.

**Comment by Valachord** [comment]
Hi, I apologize for the late reply, for some reason I thought I already posted the file.

Anyway, when I went to get the file, I saw that it is 0 bytes, which makes no sense, especially since it happened after nothing but a reboot.
I don't know if you can still use it for anything, I guess it corrupted somehow, but I will attach it just in case.

Since then, I recreated the exact same preset, I don't know if it makes any sense for you to take a look at that, but if that's the case, I will gladly post it.

[Default.zip](https://github.com/polychromatic/polychromatic/files/6532425/Default.zip)

**Comment by lah7** [comment]
Not sure what happened there, unless it was caused by an earlier crash. 🤷🏼‍♂️ 

A fix (a8b1f6dac1623b26b147be8c416b693395e6b0c1) was committed so the application doesn't throw an error over an unexpected, but harmless empty file.
